dub.sdl

```diff
@@ -49,3 +49,105 @@ subPackage {
 
   dependency "dmd:lexer" version="*"
 }
+
+subPackage {
+  name "frontend"
+  targetType "library"
+  stringImportPaths "res"
+  versions "NoBackend"
+  versions "GC"
+  versions "NoMain"
+  sourcePaths
+
+  sourceFiles \
+    "src/ddmd/access.d" \
+    "src/ddmd/aggregate.d" \
+    "src/ddmd/aliasthis.d" \
+    "src/ddmd/apply.d" \
+    "src/ddmd/argtypes.d" \
+    "src/ddmd/arrayop.d" \
+    "src/ddmd/arraytypes.d" \
+    "src/ddmd/astcodegen.d" \
+    "src/ddmd/attrib.d" \
+    "src/ddmd/builtin.d" \
+    "src/ddmd/canthrow.d" \
+    "src/ddmd/clone.d" \
+    "src/ddmd/complex.d" \
+    "src/ddmd/cond.d" \
+    "src/ddmd/constfold.d" \
+    "src/ddmd/cppmangle.d" \
+    "src/ddmd/cppmanglewin.d" \
+    "src/ddmd/ctfeexpr.d" \
+    "src/ddmd/dcast.d" \
+    "src/ddmd/dclass.d" \
+    "src/ddmd/declaration.d" \
+    "src/ddmd/delegatize.d" \
+    "src/ddmd/denum.d" \
+    "src/ddmd/dimport.d" \
+    "src/ddmd/dinifile.d" \
+    "src/ddmd/dinterpret.d" \
+    "src/ddmd/dmacro.d" \
+    "src/ddmd/dmangle.d" \
+    "src/ddmd/dmodule.d" \
+    "src/ddmd/doc.d" \
+    "src/ddmd/dscope.d" \
+    "src/ddmd/dstruct.d" \
+    "src/ddmd/dsymbol.d" \
+    "src/ddmd/dsymbolsem.d" \
+    "src/ddmd/dtemplate.d" \
+    "src/ddmd/dversion.d" \
+    "src/ddmd/escape.d" \
+    "src/ddmd/expression.d" \
+    "src/ddmd/expressionsem.d" \
+    "src/ddmd/frontend.d" \
+    "src/ddmd/func.d" \
+    "src/ddmd/hdrgen.d" \
+    "src/ddmd/id.d" \
+    "src/ddmd/impcnvtab.d" \
+    "src/ddmd/imphint.d" \
+    "src/ddmd/init.d" \
+    "src/ddmd/initsem.d" \
+    "src/ddmd/inline.d" \
+    "src/ddmd/inlinecost.d" \
+    "src/ddmd/intrange.d" \
+    "src/ddmd/json.d" \
+    "src/ddmd/lib.d" \
+    "src/ddmd/link.d" \
+    "src/ddmd/mars.d" \
+    "src/ddmd/mtype.d" \
+    "src/ddmd/nogc.d" \
+    "src/ddmd/nspace.d" \
+    "src/ddmd/objc.d" \
+    "src/ddmd/opover.d" \
+    "src/ddmd/optimize.d" \
+    "src/ddmd/parse.d" \
+    "src/ddmd/sapply.d" \
+    "src/ddmd/templateparamsem.d" \
+    "src/ddmd/semantic.d" \
+    "src/ddmd/sideeffect.d" \
+    "src/ddmd/statement.d" \
+    "src/ddmd/staticassert.d" \
+    "src/ddmd/target.d" \
+    "src/ddmd/typesem.d" \
+    "src/ddmd/traits.d" \
+    "src/ddmd/parsetimevisitor.d" \
+    "src/ddmd/visitor.d" \
+    "src/ddmd/typinf.d" \
+    "src/ddmd/utils.d" \
+    "src/ddmd/statement_rewrite_walker.d" \
+    "src/ddmd/statementsem.d" \
+    "src/ddmd/staticcond.d" \
+    "src/ddmd/safe.d" \
+    "src/ddmd/blockexit.d" \
+    "src/ddmd/printast.d" \
+    "src/ddmd/gluelayer.d"
+
+   sourceFiles "src/ddmd/scanelf.d" \
+    "src/ddmd/libelf.d" platform="linux"
+
+   sourceFiles "src/ddmd/scanmach.d" \
+    "src/ddmd/libmach.d" platform="osx"
```

Comment:
The last three files, I know they're not technically not part of the backend but I think they feel more geared towards the backend or code generation. Do we want to include them?

Reply:
Currently they are required by some bits of the frontend (see below). Though I'm pretty sure, that these parts could `version`ed out in the future.

#### Remove libelf.d

```
/tmp/.dub/build/dmd-~dub-frontend/library-debug-linux.posix-x86_64-dmd_2077-A550EF4C39A32552F8FE470A957BC323/libdmd_frontend.a(lib.o):(.data.rel.ro+0x28): undefined reference to `_D4ddmd6libelf12__ModuleInfoZ'
/tmp/.dub/build/dmd-~dub-frontend/library-debug-linux.posix-x86_64-dmd_2077-A550EF4C39A32552F8FE470A957BC323/libdmd_frontend.a(lib_532_2d5.o): In function `_D4ddmd3lib7Library7factoryFZCQBcQBaQz':
/home/seb/dlang/dmd/src/ddmd/lib.d:56: undefined reference to `LibElf_factory()'
collect2: error: ld returned 1 exit status
Error: linker exited with status 1
```

#### Remove `gluelayer.d`

```
/tmp/.dub/build/dmd-~dub-frontend/library-debug-linux.posix-x86_64-dmd_2077-A550EF4C39A32552F8FE470A957BC323/libdmd_frontend.a(dclass.o):(.data.rel.ro+0x20): undefined reference to `_D4ddmd9gluelayer12__ModuleInfoZ'
/tmp/.dub/build/dmd-~dub-frontend/library-debug-linux.posix-x86_64-dmd_2077-A550EF4C39A32552F8FE470A957BC323/libdmd_frontend.a(denum.o):(.data.rel.ro+0x10): undefined reference to `_D4ddmd9gluelayer12__ModuleInfoZ'
/tmp/.dub/build/dmd-~dub-frontend/library-debug-linux.posix-x86_64-dmd_2077-A550EF4C39A32552F8FE470A957BC323/libdmd_frontend.a(dmodule.o):(.data.rel.ro+0x28): undefined reference to `_D4ddmd9gluelayer12__ModuleInfoZ'
/tmp/.dub/build/dmd-~dub-frontend/library-debug-linux.posix-x86_64-dmd_2077-A550EF4C39A32552F8FE470A957BC323/libdmd_frontend.a(dsymbol.o):(.data.rel.ro+0x30): undefined reference to `_D4ddmd9gluelayer12__ModuleInfoZ'
/tmp/.dub/build/dmd-~dub-frontend/library-debug-linux.posix-x86_64-dmd_2077-A550EF4C39A32552F8FE470A957BC323/libdmd_frontend.a(dsymbolsem.o):(.data.rel.ro+0xd8): undefined reference to `_D4ddmd9gluelayer12__ModuleInfoZ'
/tmp/.dub/build/dmd-~dub-frontend/library-debug-linux.posix-x86_64-dmd_2077-A550EF4C39A32552F8FE470A957BC323/libdmd_frontend.a(dsymbolsem_39a_657.o): In function `Semantic3Visitor::visit(FuncDeclaration*)':
/home/seb/dlang/dmd/src/ddmd/dsymbolsem.d:1043: undefined reference to `retStyle(TypeFunction*)'
/home/seb/dlang/dmd/src/ddmd/dsymbolsem.d:1097: undefined reference to `retStyle(TypeFunction*)'
/tmp/.dub/build/dmd-~dub-frontend/library-debug-linux.posix-x86_64-dmd_2077-A550EF4C39A32552F8FE470A957BC323/libdmd_frontend.a(expression.o):(.data.rel.ro+0x40): undefined reference to `_D4ddmd9gluelayer12__ModuleInfoZ'
/tmp/.dub/build/dmd-~dub-frontend/library-debug-linux.posix-x86_64-dmd_2077-A550EF4C39A32552F8FE470A957BC323/libdmd_frontend.a(func.o):(.data.rel.ro+0x28): undefined reference to `_D4ddmd9gluelayer12__ModuleInfoZ'
/tmp/.dub/build/dmd-~dub-frontend/library-debug-linux.posix-x86_64-dmd_2077-A550EF4C39A32552F8FE470A957BC323/libdmd_frontend.a(mars.o):(.data.rel.ro+0x20): undefined reference to `_D4ddmd9gluelayer12__ModuleInfoZ'
/tmp/.dub/build/dmd-~dub-frontend/library-debug-linux.posix-x86_64-dmd_2077-A550EF4C39A32552F8FE470A957BC323/libdmd_frontend.a(mtype.o):(.data.rel.ro+0x38): undefined reference to `_D4ddmd9gluelayer12__ModuleInfoZ'
/tmp/.dub/build/dmd-~dub-frontend/library-debug-linux.posix-x86_64-dmd_2077-A550EF4C39A32552F8FE470A957BC323/libdmd_frontend.a(objc.o):(.data.rel.ro+0x60): undefined reference to `_D4ddmd9gluelayer12__ModuleInfoZ'
/tmp/.dub/build/dmd-~dub-frontend/library-debug-linux.posix-x86_64-dmd_2077-A550EF4C39A32552F8FE470A957BC323/libdmd_frontend.a(objc_5ba_3c6.o): In function `_D4ddmd4objc9Supported6__ctorMFZCQBfQBdQBb':
/home/seb/dlang/dmd/src/ddmd/objc.d:185: undefined reference to `objc_initSymbols()'
/tmp/.dub/build/dmd-~dub-frontend/library-debug-linux.posix-x86_64-dmd_2077-A550EF4C39A32552F8FE470A957BC323/libdmd_frontend.a(statement.o):(.data.rel.ro+0x30): undefined reference to `_D4ddmd9gluelayer12__ModuleInfoZ'
/tmp/.dub/build/dmd-~dub-frontend/library-debug-linux.posix-x86_64-dmd_2077-A550EF4C39A32552F8FE470A957BC323/libdmd_frontend.a(statementsem.o):(.data.rel.ro+0xd0): undefined reference to `_D4ddmd9gluelayer12__ModuleInfoZ'
/tmp/.dub/build/dmd-~dub-frontend/library-debug-linux.posix-x86_64-dmd_2077-A550EF4C39A32552F8FE470A957BC323/libdmd_frontend.a(statementsem_632_9d9.o): In function `StatementSemanticVisitor::visit(AsmStatement*)':
/home/seb/dlang/dmd/src/ddmd/statementsem.d:3923: undefined reference to `asmSemantic(AsmStatement*, Scope*)'
/tmp/.dub/build/dmd-~dub-frontend/library-debug-linux.posix-x86_64-dmd_2077-A550EF4C39A32552F8FE470A957BC323/libdmd_frontend.a(typinf.o):(.data.rel.ro+0x48): undefined reference to `_D4ddmd9gluelayer12__ModuleInfoZ'
/tmp/.dub/build/dmd-~dub-frontend/library-debug-linux.posix-x86_64-dmd_2077-A550EF4C39A32552F8FE470A957BC323/libdmd_frontend.a(aggregate.o):(.data.rel.ro+0x18): undefined reference to `_D4ddmd9gluelayer12__ModuleInfoZ'
/tmp/.dub/build/dmd-~dub-frontend/library-debug-linux.posix-x86_64-dmd_2077-A550EF4C39A32552F8FE470A957BC323/libdmd_frontend.a(typinf_680_468.o): In function `genTypeInfo(Type*, Scope*)':
/home/seb/dlang/dmd/src/ddmd/typinf.d:67: undefined reference to `toObjFile(Dsymbol*, bool)'
```

Reply:
I see. I think we can do some refactoring later to remove these dependencies if we want to.

BTW, I think `lib.d` is for generating libraries, which I don't think we need.